src/guest_memory.rs

```diff
@@ -386,6 +386,21 @@ impl<T: GuestMemory> Bytes<GuestAddress> for T {
         )
     }
 
+    /// # Examples
+    /// * Write a slice at guestaddress 0x200.
+    ///
+    /// ```
+    /// # #[cfg(feature = "backend-mmap")]
+    /// # fn test_write_u64() -> Result<(), ()> {
+    /// # use vm_memory::{Bytes, GuestAddress, mmap::GuestMemoryMmap};
```

Comment:
Does the `use` need to live inside the `fn`?

Reply:
Let's not use fn in examples. It's easier to keep the code and examples in sync if we don't have functions because failures are caught by the CI. If you just return an Error when something goes wrong and don't check the error, failures go unnoticed.

Reply:
This also applies to existing tests it would be nice if you could remove the use of `fn`, for example:

```diff
diff --git a/src/volatile_memory.rs b/src/volatile_memory.rs
index 1883c46..9178783 100644
--- a/src/volatile_memory.rs
+++ b/src/volatile_memory.rs
@@ -323,17 +323,14 @@ impl<'a> VolatileSlice<'a> {
     /// # use std::fs::File;
     /// # use std::path::Path;
     /// # use vm_memory::VolatileMemory;
-    /// # fn test_write_null() -> Result<(), ()> {
     /// let mut mem = [0u8; 32];
     /// let mem_ref = &mut mem[..];
-    /// let vslice = mem_ref.get_slice(0, 32).map_err(|_| ())?;
+    /// let vslice = mem_ref.get_slice(0, 32).unwrap();
     /// let mut buf = [5u8; 16];
     /// vslice.copy_to(&mut buf[..]);
     /// for v in &buf[..] {
     ///     assert_eq!(buf[0], 0);
     /// }
-    /// # Ok(())
-    /// # }
     /// ```
     pub fn copy_to<T>(&self, buf: &mut [T]) -> usize
     where
```

(but be careful with tests that try to open /dev/urandom).

Reply:
I used fn in order to be able to specify `#[cfg(feature = "backend-mmap")]` since this is needed for using `GuestMemoryMmap` . I couldn't think of a better solution for this, but I did some changes.